### DISCUSSION TITLE
Dynamically Alter Text of Current Node

### DISCUSSION THREAD

**Discussion by ThunderSloth** [L3MON4D3/LuaSnip#1076] [Q&A] [answered]
I'm trying to create a node that if anything is entered it wraps the text in [square brackets] but if its left blank than it does not. How would I go about doing this? Here is my latest try:
```	
        s( -- root
		{trig = "\\sq", snippetType = "autosnippet",},
		fmta(
			[[\sqrt<>{<>} <>]],
			{
				d(1, function(args)
					local text = args[1]
					if text ~= "" then
						text = string.format("[%s]", text)
					end
					return sn(nil, {
						i(1, text)
					})
				end),
				i(2), 
				i(3), 
			}
		)
	}),
```
This doesn't work, however. When the snippet is expanded it gives: ﻿﻿\sqrt[nil]{} 
As soon as I start typing the [nil] goes away, but of course I'm left with a normal insert node, so it doesn't do what I want.

**Comment by L3MON4D3** [comment]
I think that's a good usecase for jump-events (don't have a link at hand, but searching for events in DOC.md should yield some results).
You'd have to register a function that retrieves the node using API (don't have it at hand, but I'm pretty sure there is something like get_keyed_node which can retrieve nodes that have a key), the only missing piece is that we don't yet have an official API for setting text, but I think node:set_text will work (at least for the foreseeable future)
If you'd like more details, I can look into whipping up a quick example in a few days when I have more time :D

**Reply by ThunderSloth** [reply]
I couldn't find anything on "jump-events" but I was able to solve my problem by surrounding the input node with function nodes, although after looking over the doc again I think notext nodes would be an even simpler solution.

**Reply by ThunderSloth** [reply]
They're called "nonempty" nodes not "notext", I misremembered. But anyways, they work perfectly here.
```
	s( -- root
		{trig = "sq", snippetType = "autosnippet",},
		fmta(
			[[\sqrt<><><>{<>} <>]],
			{
				extras.nonempty(1, "[", ""),
				d(1, utils.get_choices, {}, {user_args = {utils.defaults.root}}),
				extras.nonempty(1, "]", ""),
				i(2), 
				i(3), 
			}
		)
	),

```

**Reply by L3MON4D3** [reply, marked as answer]
Ah, yeah using a functionNode makes much more sense :facepalm: 
Also true, good usecase for `nonempty` :+1:
